ballerina-tests/tests/hateoas_runtime_error_tests.bal

```diff
@@ -0,0 +1,132 @@
+import ballerina/http;
+import ballerina/test;
+
+listener http:Listener hateoasRuntimeErrorServerEP = check new(hateoasRuntimeErrorPort);
+
+@test:Config{}
+function testDuplicateResourceName() {
+    http:Service duplicateResourceNameService = service object {
+        @http:ResourceConfig{
+            name: "resource1"
+        }
+        resource function get resource1() returns string {
+            return "resource1";
+        }
+
+        @http:ResourceConfig{
+            name: "resource1"
+        }
+        resource function post resource2() returns string {
+            return "resource2";
+        }
+    };
+    error? err = hateoasRuntimeErrorServerEP.attach(duplicateResourceNameService, "service1");
+    if err is error {
+        test:assertEquals(err.message(), "service registration failed: cannot duplicate resource name:" +
```

Comment:
Here by the resource name, we represent the name field in the http:ResourceConfig. But same resource name is used for the resource path. Need to fix this ambiguity 

https://github.com/ballerina-platform/module-ballerina-http/blob/master/docs/spec/spec.md#232-resource-name

Reply:
Can we say `cannot duplicate resource link name:`?

Reply:
+1 for `resource link name`

Reply:
Fixed in [9b1a15c](https://github.com/ballerina-platform/module-ballerina-http/pull/1068/commits/9b1a15ce791c2179473d67578d1acbb8c934d313)